test/widgets/message_list_test.dart

```diff
@@ -136,20 +136,20 @@ void main() {
       check(state.narrow).equals(ChannelNarrow(stream.streamId));
     });
 
-    testWidgets('composeBoxController finds compose box', (tester) async {
+    testWidgets('composeBoxState finds compose box, with controller', (tester) async {
       final stream = eg.stream();
       await setupMessageListPage(tester, narrow: ChannelNarrow(stream.streamId),
         streams: [stream],
         messages: [eg.streamMessage(stream: stream, content: "<p>a message</p>")]);
       final state = MessageListPage.ancestorOf(tester.element(find.text("a message")));
-      check(state.composeBoxController).isNotNull();
+      check(state.composeBoxState?.controller).isNotNull();
```

Comment:
Similar to the earlier comment on a null-check, I feel that checking if `composeBoxState?.controller` is null will make more sense if the controller is nullable. The updated contract seems to be: as long as `composeBoxState` is present, there is a compose box and vice versa.

Reply:
Yeah; I think what happened is I did something like a copy-paste and didn't make this refinement from there :)